packages/core/src/scripts/migrate-oz-cli-project.ts

```diff
@@ -0,0 +1,434 @@
+    src: storageItem.path,
+  };
+}
+
+function transformTypes(oldTypes: LegacyTypes): Record<string, TypeItem> {
+  const newTypes: Record<string, TypeItem> = {};
+  for (const typeName in oldTypes) {
+    newTypes[transformTypeName(typeName)] = transformType(getTypeKind(typeName), oldTypes[typeName]);
+  }
+  return newTypes;
+}
+
+function transformType(typeKind: TypeKind, oldType: LegacyType): TypeItem {
+  switch (typeKind) {
+    case 'Struct':
+      return {
+        label: stripContractName(oldType.label),
+        members: (oldType.members as StructMember[]).map(member => ({
+          label: stripContractName(member.label),
+          type: transformTypeName(member.type),
+        })),
+      };
+    case 'Enum':
+      return {
+        label: stripContractName(oldType.label),
+        members: oldType.members,
+      };
+    default:
+      return {
+        label: oldType.label,
+      };
+  }
+}
+
+function transformTypeName(typeName: string): string {
+  switch (getTypeKind(typeName)) {
+    case 'Struct':
+      return transformStructTypeName(typeName);
+    case 'Enum':
+      return transformEnumTypeName(typeName);
+    case 'Mapping':
+      return transformMappingTypeName(typeName);
+    case 'DynArray':
+      return transformDynArrayTypeName(typeName);
+    case 'StaticArray':
+      return transformStaticArrayTypeName(typeName);
+    case 'Elementary':
+      return typeName;
+    default:
+      throw new Error(`Unknown type: ${typeName}`);
+  }
+}
+
+function transformStructTypeName(typeName: string): string {
+  const name = stripContractName(getArgument(typeName));
+  return `t_struct(${name})_storage`;
+}
+
+function transformEnumTypeName(typeName: string): string {
+  const name = stripContractName(getArgument(typeName));
+  return `t_enum(${name})`;
+}
+
+function transformMappingTypeName(typeName: string): string {
+  const valueType = transformTypeName(getArgument(typeName));
+  return `t_mapping(unknown,${valueType})`;
+}
+
+function transformDynArrayTypeName(typeName: string): string {
+  const valueType = transformTypeName(getArgument(typeName));
+  return `t_array(${valueType})dyn_storage`;
+}
+
+function transformStaticArrayTypeName(typeName: string): string {
+  // here we assume the regex has been already validated
+  // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+  const size = typeName.match(/:(\d+)/)![1];
+  const valueType = transformTypeName(getArgument(typeName));
+  return `t_array(${valueType})${size}_storage`;
+}
+
+function getArgument(typeName: string): string {
+  // here we assume the regex has been already validated
+  // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+  return typeName.match(/<(.+)>/)![1];
+}
+
+function stripContractName(s: string): string {
+  const match = s.match(/(.+)\.(.+)/);
+  // input might not contain a contract name, so we fallback to it
+  return match ? match[2] : s;
```

Comment:
I think this is better expressed by an optional regex group.

```suggestion
  // regex always matches
  // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
  return s.match(/(.+\.)?(.+)/)![2];
```